kusd/validator/states.go

```diff
@@ -91,6 +52,60 @@ func (val *Validator) notLoggedInState() stateFn {
 	return val.newElectionState
 }
 
+func (val *Validator) nodeNeedsToDeposit() bool {
+	// @NOTE (rgeraldes) - sync was already done at this point and by default the investors will be
+	// part of the initial set of validators - no need to make a deposit if the block number is 0
+	// since these validators will be marked as voters from the start
+	isGenesis, err := val.network.IsGenesisVoter(&bind.CallOpts{}, val.account.Address)
+	if err != nil {
+		log.Crit("Failed to verify the voter information", "err", err)
+	}
+	return !isGenesis || val.isGenesisRejoining(isGenesis)
+}
+
+func (val *Validator) isGenesisRejoining(isGenesis bool) bool {
+	return isGenesis && val.isNotFirstBlock()
+}
+
+func (val *Validator) isFirstBlock() bool {
+	return val.chain.CurrentBlock().NumberU64() == 0
+}
+
+func (val *Validator) isNotFirstBlock() bool {
+	return !val.isFirstBlock()
+}
+
+func (val *Validator) waitToParticipate() {
+	// Subscribe events from blockchain
+	chainHeadCh := make(chan core.ChainHeadEvent)
+	chainHeadSub := val.chain.SubscribeChainHeadEvent(chainHeadCh)
+	defer chainHeadSub.Unsubscribe()
+
+	log.Info("Waiting confirmation to participate in the consensus")
+L:
+	for {
+		select {
+		case _, ok := <-chainHeadCh:
+			if !ok {
+				log.Info("Failed reading on Header channel")
+			}
+
+			isVoter := val.assertGetIsVoter()
+			if isVoter {
+				break L
```

Comment:
I think now we can just `return` and not use labels.

Reply:
It's been year's since I've seen Gotos :) happy you guys wanna ditch that. I didn't here, because didn't went thru the logic, was just splinting in smaller chunks